src/runner/Run_semgrep.ml

```diff
@@ -674,9 +675,23 @@ let semgrep_with_rules config ((rules, invalid_rules), rules_parse_time) =
              }
            in
            let matches =
-             Match_rules.check ~match_hook ~timeout:config.timeout
-               ~timeout_threshold:config.timeout_threshold xconf rules xtarget
+             let matches =
+               Match_rules.check ~match_hook ~timeout:config.timeout
+                 ~timeout_threshold:config.timeout_threshold xconf rules xtarget
+             in
+             (* If our target is a proprietary language, or we've been using the proprietary
+              * engine, then label all the resulting matches with the Pro engine kind.
+              * This can't really be done any later, because we need the language that
+              * we're running on.
+              *)
+             if config.pro || Xlang.is_proprietary xtarget.xlang then
```

Comment:
Feels wrong to have an `engine_kind` in `Pattern_match.t` and then use it this way.

Reply:
I don't have a strong opinion about having to change this now, but long-term I think we either change how `engine_kind` is decided, or it shouldn't be part of `Pattern_match.t`.

Reply:
Yeah, I agree, but for data-dependency and type-related reasons, I found this really difficult to do when originally setting it up.

I think this will make more sense once we implement tracking whether a match resulted specifically from Pro (as in, it couldn't be found with OSS)